src/Microsoft.TemplateEngine.Cli/ChoiceTemplateParameter.cs

```diff
@@ -184,29 +186,46 @@ private static bool TryConvertValueToChoice(IEnumerable<string> values, ChoiceTe
                 parsedValues.Add(value);
             }
 
+            // Single empty value allowed as explicit unset
```

Comment:
nit: a bit confusing comment. the next section is about error when empty value is specified with something else.
```suggestion
            // An empty value is not allowed when multiple choice values are specified.
```

On same side: does parser allow 2 or more values which are the same? 
i.e.: `--platform iOS iOS` or `--platform iOS --platform iOS`

Reply:
This is interesting idea - it is allowed and can be easily flagged in this function. However I would not do that - I believe it should be allowed to repeate same option, as it's possible in other command line tools - e.g. `del f1.txt f2.txt f1.txt`
(This is more friendly to automated scripting as well)